R/mrgindata.R

```diff
@@ -81,45 +81,62 @@ convert_character_cmt <- function(data, mod) {
   return(data)
 }
 
-signal_drop <- function(dm, x, to_signal, context) {
-  drop <- setdiff(names(x), dimnames(dm)[[2]])
-  drop <- intersect(drop, to_signal)
-  for(d in drop) {
+check_dropped_cols <- function(dm, x, check, context) {
+  new_col_set <- dimnames(dm)[[2]]
+  old_col_set <- names(x)
+  drop <- old_col_set[!old_col_set %in% new_col_set]
+  drop <- drop[drop %in% check]
+  if(!length(drop)) return(invisible(NULL))
+  body <- vector(mode = "character", length = length(drop))
+  names(body) <- rep("x", length(body))
+  for(i in seq_along(drop)) {
+    d <- drop[i]
     type <- paste0(class(x[[d]]), collapse = ",")
-    msg <- c(context, " dropped column: ", d, " (", type, ")")
-    message(msg)  
+    body[i] <- paste0(context, " column: ", d, " (", type, ")")
   }
-  invisible(NULL)
+  abort(
+    message = "Found input data that cannot be used for simulation", 
+    body = body, 
+    call = caller_env()
+  )
 }
 
 ##' Validate and prepare a data sets for simulation
```

Comment:
Nearby typo in case you want to fix it up here: should be either "prepare data sets" (that's consistent with `valid_idata_set`) or "prepare a data set".

Reply:
Thanks, @kyleam . FIxed now.